src/main/java/pl/tomaszdziurko/jvm_bloggers/view/login/attack/BruteForceAttackMailGenerator.java

```diff
@@ -0,0 +1,49 @@
+package pl.tomaszdziurko.jvm_bloggers.view.login.attack;
+
+import com.google.common.base.Preconditions;
+import lombok.extern.slf4j.Slf4j;
+import org.antlr.stringtemplate.StringTemplate;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+import pl.tomaszdziurko.jvm_bloggers.utils.NowProvider;
+
+import java.time.format.DateTimeFormatter;
+
+/**
+ * @author Adam Dec
+ * @since 0.7.0
+ */
+@Component
+@Slf4j
+public class BruteForceAttackMailGenerator {
+
+    private static final String TIME = "Time";
```

Comment:
No, only constructions like that looks a bit odd for me:
````java
template.setAttribute(TIME, nowProvider.now().format(FORMATTER));
````

For me this code is over-engineered. I would write this method like that:
````java
public String prepareMailContent(BruteForceAttackEvent bruteForceAttackEvent) {
        return String.format(
            "Brute-force attack detected from IP: %s at %s",
            bruteForceAttackEvent.getIpAddress(),
            nowProvider.now().format(DateTimeFormatter.ofPattern("HH:mm:ss"))
        );
    }
````